[COLIB-NGC] add kundur two area test system (#214)
CRESYM/colib0.github.io: pages/testCases/kundur_two_area_system/index.md

| G4             | $$700.000 + \mathrm{j}202.071$$ | $$700 + \mathrm{j}202$$       |

- The active power exchange from Area 1 to Area 2 (see
  <a href="#fig-two-area-system" class="quarto-xref">Figure 1</a>) is
  400.431 MW in the Dyna$$\omega$$o implementation and 400 MW in the
  Kundur reference.

### Example a

*Example a* examines small-signal performance of the system with
constant excitation.

- All four generators are on manual excitation (field voltage
  $$E_\mathrm{fd} = \mathrm{const.}$$)
- Constant current characteristics of active load components
- Constant impedance characteristics of reactive load components

<a href="#fig-example-a-eigenvalues" class="quarto-xref">Figure 2</a>
shows the result of an eigenvalue analysis of this system, compared to
the reference values from [1]. The remaining differences probalby
originate from differing model implementations of the synchronous
machines in Dyna$$\omega$$ compared to the ones used by Kundur.

![](images/20240814123344.png)

<div id="fig-example-a-eigenvalues">

Figure 2: Eigenvalues of the system with constant excitation voltage

</div>

> ### Example b

Reviewer: Which one is the example b? Perhaps ad one sentence like in example a?

> [!NOTE]
>
> To obtain a steady state initial condition, the exact values for the
> generators active and reactive power and voltage angles from load flow
> (<a href="#sec-loadflow" class="quarto-xref">Section 6.1</a>) need to
> be used as initial values in the dynamic simulation.

> [!NOTE]
>
> In the Dyna$$\omega$$o implementation, if the voltage transducer time
> constant $$T_\mathrm{R}$$ = 0.01 s, then the voltage measurement first
> order lag and the field flux linkage $$\lambda_\mathrm{f}$$ oscillate
> against each other with high amplitude. This is not represented in the
> reference results of [2]. To avoid this behaviour, set

> > $$T_\mathrm{R} = 10^{-6}$$ s. This has been done for the Dyna$$\omega$$o

Reviewer: This really requires some investigation. In practice you say that the filter/transducer was removed from the model (which 
 model by the way? ST1C?)

> implementation.

#### Part 1: Self-excieted dc exciter (DC1A)

Instead of constant excitation voltage as in
<a href="#sec-exampleA" class="quarto-xref">Section 6.2</a>, now the
self-excited DC exciter DC1A is used. For parameters, see
<a href="#tbl-dc1a-parameters" class="quarto-xref">Table 3</a>.

In [2] this case has not been taken into account for comparative
analysis. Like there, it has not been investigated using the
Dyna$$\omega$$o implementation.

#### Part 2: Thyristor exciter (ST1A) with high transient gain

Instead of constant excitation voltage as in
<a href="#sec-exampleA" class="quarto-xref">Section 6.2</a>, now the
thyristor exciter ST1A is used. For parameters, see
<a href="#tbl-dc1a-parameters" class="quarto-xref">Table 3</a>. It has
high transient gain, i.e. no transient gain reduction.

> The implementation in Dyna$$\omega$$o does not allow to set saturation

Reviewer: Is it possible that the saturation is the cause of the difference between the Kundur and Dynawo eigenvalues? How do the other eigenvalues move when trying to match the unstable one?
Instead of matching the unstable eigenvalue perhaps try to match as closely as possible the Kundur and Shackshaft saturation curves?

parameters $$A_\mathrm{sat}$$ and $$B_\mathrm{sat}$$ as in
<a href="#tbl-sm-parameters" class="quarto-xref">Table 2</a> because its
synchronous machine model uses a different saturation model. Instead of
the model from Kundur [1], the Shackshaft model [3] is used. To get

> the unstable pole close to the one in the Kundur reference, the

Reviewer: ```suggestion
the unstable (pole or  eigenvalue?) close to the one in the Kundur reference, the
```

Shackshaft parameters have been set to $$m_\mathrm{d}=1$$,
$$m_\mathrm{q}=1$$, $$n_\mathrm{d}=9$$, $$n_\mathrm{q}=9$$.

> The voltage transducer time constant has been

Reviewer: Unfinished sentence

![](images/20240814131709.png)

<div id="fig-example-b-2-eigenvalues">

Figure 3: Unstable eigenvalue of the system with exciter ST1A

</div>

> As in [2], a steps to the synchronous machines’ voltege references

Reviewer: ```suggestion
As in [2], steps to the synchronous machines’ voltage references
```

have been applied according to
<a href="#tbl-example-b-steps" class="quarto-xref">Table 11</a>. Those
were selected to excite the inter-area oscillation mode and the other
electromechanical modes in the system [2].

<div id="tbl-example-b-steps">

Table 11: Voltage reference steps as in [2]
</div>

| Generator | Step of $$U_\mathrm{ref}$$ |
| :-------- | :------------------------- |
| G1        | +3%                        |
| G2        | -1%                        |
| G3        | -3%                        |
| G4        | +1%                        |

The step response of the generators can be seen in
<a href="#fig-example-b-2-stepresponse" class="quarto-xref">Figure 4</a>.

![](images/20240814133617.png)

<div id="fig-example-b-2-stepresponse">

Figure 4: Unstable step response of the system with ST1A without
transient gain reduction

</div>

#### Part 3: Thyristor exciter (ST1A) with transient gain reduction (TGR)

> Simlilarly to

Reviewer: ```suggestion
Similarly to
```

<a href="#sec-exampleB2" class="quarto-xref">Section 6.3.2</a>, the ST1A
exciter is used. In this case, transient gain reduction (TGR) is
activated by setting $$T_\mathrm{A} = 1$$.

#### Part 4: Thyristor exciter (ST1A) with high transient gain and PSS

> Simlilarly to

Reviewer: ```suggestion
Similarly to
```

<a href="#sec-exampleB2" class="quarto-xref">Section 6.3.2</a>, the ST1A
exciter is used. In this case, a power system stabilizer according to
[1] (Figure 6-8) is activated and parameterized according to
<a href="#tbl-pss-parameters" class="quarto-xref">Table 5</a>.

## Open source implementations

This model has been successfully implemented in :

| Software               | URL                                        | Language | Open-Source License                                | Last consulted date | Comments |
| ---------------------- | ------------------------------------------ | -------- | -------------------------------------------------- | ------------------- | -------- |

> | Open Modelica / Dynawo | [Dynawo](https://github.com/dynawo/dynawo) | modelica | [MPL v2.0](https://www.mozilla.org/en-US/MPL/2.0/) | 15/08/2024          | \-       |

Reviewer: Is there a link directly to the model? I see only one to the modelica github
